src/analyzer/analyzersilence.cpp

```diff
@@ -0,0 +1,104 @@
+#include "analyzer/analyzersilence.h"
+
+const int kChannelCount = 2;
+const float kSilenceThreshold = -60.0f;
+
+AnalyzerSilence::AnalyzerSilence(UserSettingsPointer pConfig)
+    : m_pConfig(pConfig),
+      m_fThreshold(db2ratio(kSilenceThreshold)),
+      m_iSampleRate(0),
+      m_iTotalSamples(0),
+      m_iFramesProcessed(0),
+      m_bPrevSilence(true),
+      m_iSignalBegin(-1),
+      m_iSignalEnd(-1) {
+}
+
+AnalyzerSilence::~AnalyzerSilence() {
+}
+
+bool AnalyzerSilence::initialize(TrackPointer tio, int sampleRate, int totalSamples) {
+    Q_UNUSED(tio);
+
+    m_iSampleRate = sampleRate;
+    m_iTotalSamples = totalSamples;
+    m_iFramesProcessed = 0;
+    m_bPrevSilence = true;
+    m_iSignalBegin = 0;
+    m_iSignalEnd = 0;
+
+    return true;
+}
+
+bool AnalyzerSilence::isDisabledOrLoadStoredSuccess(TrackPointer tio) const {
+    Q_UNUSED(tio);
+
+    return false;
+}
+
+void AnalyzerSilence::process(const CSAMPLE *pIn, const int iLen) {
+    for (int i = 0; i < iLen; i += kChannelCount) {
```

Comment:
Ah sorry, I have misread the code. It just detects the first and last sample above -60 dB. 
So it should work as it. 

However, It will fail if we have click sounds in the lead-in and lead out. Do we need to consider this?